examples/python/trigger-pagerduty-incident/handler/handler.py

```diff
@@ -0,0 +1,143 @@
+    """    
+    def __init__(self, status, message):
+        """
+        
+        Arguments:
+
+            status {str} -- the response status code
+            message {str} -- the response message
+        """        
+        self.status=status
+        self.message=message
+
+class Pagerduty:
+    """Pagerduty is a Class used to trigger Events in PagerDuty."""    
+
+    def __init__(self,conn):
+        """
+        
+        Arguments:
+
+            conn {Session} -- connection to vCenter REST API
+        """        
+
+        try:
+             with open(PD_CONFIG, 'r') as pdconfigfile:
+                pdconfig = json.load(pdconfigfile)
+                self.routingkey=pdconfig['routing_key']
+                self.event_action=pdconfig['event_action']
+        except OSError as e:
+            print(f'could not read PagerDuty configuration: {e}')
+            sys.exit(1)
+        except KeyError as e:
+            print(f'mandatory configuration key not found: {e}')
+            sys.exit(1)
+        self.session=conn
+
+    # PagerDuty REST API implementation        
+    def connect(self,obj):
+        """make a rest api call to Pagerduty Events API
+        
+        Arguments:
+
+            obj {dict} -- Generated API Body for the PagerDuty Events API
+
+        Returns:
+            FaaSResponse -- status code and message
+        """
+        obj['routing_key'] = self.routingkey
+        obj['event_action'] = self.event_action
+        try:
+            resp = self.session.post(PAGERDUTY_API_PATH,json=obj)
+            resp.raise_for_status()
+            print(resp.text)
+            resp_body = json.loads(resp.text)
+            return FaaSResponse('200', 'successfully {0}ed event action with dedup_key: {1}'.format(self.event_action, resp_body['dedup_key']))
+        except requests.HTTPError as err:
+            return FaaSResponse('500', 'could not invoke Pagerduty API {0}'.format(err))
+
+def handle(req):
+    # Validate input
+    try:
+        ref = json.loads(req)
+    except ValueError as err:
+        res = FaaSResponse('400','invalid JSON {0}'.format(err))
+        print(json.dumps(vars(res)))
+        return
+    
+    # Assert that the Event conforms to CloudEvent Spec
+    # For debugging: validate the JSON blob we received - uncomment if needed
+    # print(j)
+    try:
+        eventdata = (ref['data'])
+    except KeyError as err:
+        res = FaaSResponse('400','JSON does not conform to CloudEvent spec {0}'.format(err))
+        print(json.dumps(vars(res)))
+        return
+
+    # Map the CloudEvent data and build the PagerDuty Event API Request body
+    obj = {
+            'payload': {
+                'summary': eventdata['FullFormattedMessage'],
+                'timestamp': eventdata['CreatedTime'],
+                'source': ref['source'],
+                'severity': 'info',
+                'component': eventdata['Vm']['Name'],
```

Comment:
What if this key does not exist, e.g. someone subscribed to the wrong event. It's probably a good idea to check for these fields (Vm, Host) in the event payload received with try/except and handle incorrect gracefully (with appropriate error message).

Reply:
https://code.vmware.com/apis/704/vsphere/vim.event.Event.html - Does this mean that there are some event attributes that are generated with every event? Can i skip validation for the required fields noted for the Event Object?